### PR TITLE
feat/roe-2669: retrieve application data from DB for the `trust-details` page

### PR DESCRIPTION
**Pull request by mwejuli-ch**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-2669

### Change description

- Retrieves application data from the API for the `trust-details` page/screen
- Modifies tests to match new/updated functionality
- Includes minor formatting changes

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### CODE REVIEW

**Review by markpit — APPROVED** [review]

**Comment by mwestacott** [comment]
> **SonarQube Quality Gate**
> 
> [Quality Gate passed](http://code-analysis.platform.aws.chdev.org:9000/dashboard?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665)
> 
> [Bug](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [0 Bugs](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [Vulnerability](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [Security Hotspot](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [A](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [0 Security Hotspots](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [Code Smell](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL) [1 Code Smell](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL)
> 
> [](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_coverage&view=list) [100.0% Coverage](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_coverage&view=list) [](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_duplicated_lines_density&view=list)

The code smell should ideally be addressed.

**Comment by markpit** [comment]
> > **SonarQube Quality Gate**
> > [Quality Gate passed](http://code-analysis.platform.aws.chdev.org:9000/dashboard?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665)
> > [Bug](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [0 Bugs](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [Vulnerability](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [Security Hotspot](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [A](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [0 Security Hotspots](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [Code Smell](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL) [1 Code Smell](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL)
> > [](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_coverage&view=list) [100.0% Coverage](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_coverage&view=list) [](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_duplicated_lines_density&view=list)
> 
> The code smell should ideally be addressed.

I did see it but it would need a refactor of the function, and moses is just adding an if/else which is needed for the redis removal so thought it wouldn't be too much of an issue.

**Comment by mwestacott** [comment]
> > > **SonarQube Quality Gate**
> > > [Quality Gate passed](http://code-analysis.platform.aws.chdev.org:9000/dashboard?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665)
> > > [Bug](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [0 Bugs](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [Vulnerability](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [Security Hotspot](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [A](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [0 Security Hotspots](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [Code Smell](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL) [1 Code Smell](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL)
> > > [](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_coverage&view=list) [100.0% Coverage](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_coverage&view=list) [](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_duplicated_lines_density&view=list)
> > 
> > 
> > The code smell should ideally be addressed.
> 
> I did see it but it would need a refactor of the function, and moses is just adding an if/else which is needed for the redis removal so thought it wouldn't be too much of an issue.

Yep, I suppose given it's already quite long and complex we can live with this. Next time it changes though...

**Comment by mwejuli-ch** [comment]
> > > > **SonarQube Quality Gate**
> > > > [Quality Gate passed](http://code-analysis.platform.aws.chdev.org:9000/dashboard?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665)
> > > > [Bug](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [0 Bugs](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [Vulnerability](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [Security Hotspot](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [A](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [0 Security Hotspots](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [Code Smell](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL) [1 Code Smell](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL)
> > > > [](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_coverage&view=list) [100.0% Coverage](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_coverage&view=list) [](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_duplicated_lines_density&view=list)
> > > 
> > > 
> > > The code smell should ideally be addressed.
> > 
> > 
> > I did see it but it would need a refactor of the function, and moses is just adding an if/else which is needed for the redis removal so thought it wouldn't be too much of an issue.
> 
> Yep, I suppose given it's already quite long and complex we can live with this. Next time it changes though...

have actually refactored the method to fix this.. looks to be all good now

**Comment by mwestacott** [comment]
> > > > > **SonarQube Quality Gate**
> > > > > [Quality Gate passed](http://code-analysis.platform.aws.chdev.org:9000/dashboard?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665)
> > > > > [Bug](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [0 Bugs](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=BUG) [Vulnerability](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=VULNERABILITY) [Security Hotspot](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [A](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [0 Security Hotspots](http://code-analysis.platform.aws.chdev.org:9000/security_hotspots?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665) [Code Smell](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL) [A](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL) [1 Code Smell](http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&resolved=false&types=CODE_SMELL)
> > > > > [](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_coverage&view=list) [100.0% Coverage](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_coverage&view=list) [](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](http://code-analysis.platform.aws.chdev.org:9000/component_measures?id=uk.gov.companieshouse%3Aoverseas-entities-web&pullRequest=1665&metric=new_duplicated_lines_density&view=list)
> > > > 
> > > > 
> > > > The code smell should ideally be addressed.
> > > 
> > > 
> > > I did see it but it would need a refactor of the function, and moses is just adding an if/else which is needed for the redis removal so thought it wouldn't be too much of an issue.
> > 
> > 
> > Yep, I suppose given it's already quite long and complex we can live with this. Next time it changes though...
> 
> have actually refactored the method to fix this.. looks to be all good now

Nice one! 😄

**Review by mwestacott — APPROVED** [review]

File: src/utils/trust.details.ts [hunk 1]
```diff
@@ -169,67 +195,68 @@ export const postTrustDetails = async (req: Request, res: Response, next: NextFu
...
     return safeRedirect(res, getNextPage(isUpdate, details.trust_id, req, isReview));
+
   } catch (error) {
     logger.errorRequest(req, error);
-
     return next(error);
   }
 };
 
+const getAndUpdateTrustData = (req: Request, isReview: boolean, appData: ApplicationData, details: Trust) => {
```
Comment (lduranteau) [inline src/utils/trust.details.ts:239]: the name of the parameter details seems odd to me because in reality it is a Trust

Comment (mwejuli-ch) [reply src/utils/trust.details.ts]: have now changed it to `trustDetails`..

**Review by lduranteau — APPROVED** [review]

**Pull request merged**